api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -512,13 +517,41 @@ public void setDosing(DosingInstructions di) {
 	 */
 	@Override
 	public boolean hasSameOrderableAs(Order otherOrder) {
-		if (!super.hasSameOrderableAs(otherOrder)) {
-			return false;
-		}
-		if (!(otherOrder instanceof DrugOrder)) {
-			return false;
-		}
-		DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
-		return OpenmrsUtil.nullSafeEquals(this.getDrug(), otherDrugOrder.getDrug());
+        if (!super.hasSameOrderableAs(otherOrder)) {
+            return false;
+        }
+        if (!(otherOrder instanceof DrugOrder)) {
+            return false;
+        }
+        DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
+
+        if (isNonCodedDrug() || otherDrugOrder.isNonCodedDrug()) {
```

Comment:
But other code will fail before the validator is called.
If both drug orders have null drug and null non coded values, i think it is fair to think they are the same and it is okay for the condition to fail.